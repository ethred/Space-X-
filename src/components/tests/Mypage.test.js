import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MyPage from '../Mypage';

const mockStore = configureStore([]);
const initialState = {
  missions: {
    missions: [
      { id: 1, mission_name: 'Mission 1', joined: true },
      { id: 2, mission_name: 'Mission 2', joined: false },
    ],
  },
  rockets: {
    rockets: [
      { id: 1, name: 'Rocket 1', reserved: true },
      { id: 2, name: 'Rocket 2', reserved: false },
    ],
  },
};
const store = mockStore(initialState);

test('MyPage component matches snapshot', () => {
  const { container } = render(
    <Provider store={store}>
      <MyPage />
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
