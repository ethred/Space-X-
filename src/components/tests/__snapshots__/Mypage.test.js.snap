// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPage component matches snapshot 1`] = `
<div>
  <section>
    <div
      class="type"
    >
      <h2>
        Rocket Joined
      </h2>
      <ul>
        <li>
          Rocket 1
        </li>
      </ul>
    </div>
    <div
      class="type"
    >
      <h2>
        Missions Joined
      </h2>
      <ul>
        <li>
          Mission 1
        </li>
      </ul>
    </div>
  </section>
</div>
`;
